### PR TITLE
Refactor the package

### PR DESCRIPTION
**Pull request by koher**
Separates the packages into the following subpackages.

- `SwiftScript`
- `SwiftAST`
- `SwiftParse`

### CODE REVIEW

File: Sources/SwiftParse/TopLevel.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import SwiftAST
+import Runes
+import TryParsec
+import Result
+
+let topLevel: SwiftParser<[Statement]> = stmtBraceItems <* OWS <* endOfInput()
+
+public func parse(_ sourceCode: String) throws -> [Node] {
```
Comment (rintaro) [inline Sources/SwiftParse/TopLevel.swift]: Why you changed this from `[Statement]` to `[Node]`?

File: Package.swift [hunk 2]
```diff
@@ -2,6 +2,11 @@ import PackageDescription
 
 let package = Package(
     name: "SwiftScript",
+    targets: [
+        Target(name: "SwiftAST"),
+        Target(name: "SwiftParse", dependencies: ["SwiftAST"]),
+        Target(name: "SwiftScript", dependencies: ["SwiftParse"]),
```
Comment (rintaro) [inline Package.swift]: I suggest adding `"SwiftAST"` to`SwiftScript` dependencies as well.
Although it's not necessary, sources in `SwiftScript` directly import `SwiftAST`.

File: Sources/SwiftAST/Visitors.swift [hunk 3]
```diff
@@ -279,6 +279,3 @@ extension ImplicitlyUnwrappedOptionalType {
 extension ProtocolCompositionType {
     public func accept<V : TypeVisitor>(_ v: V) throws -> V.TypeResult { return try v.visit(self) }
 }
-
-public protocol ASTVisitor: DeclarationVisitor, ExpressionVisitor, StatementVisitor, TypeVisitor {
-}
```
Comment (rintaro) [inline Sources/SwiftAST/Visitors.swift:284]: I agree we don't need this for now 👍

File: Sources/SwiftParse/TopLevel.swift (hunk 1 above)
Comment (koher) [reply Sources/SwiftParse/TopLevel.swift]: Hmm. I though top level nodes can contain something other than a statement. However top level nodes must be statements. I'll revert it to return `[Statement]`.

File: Package.swift (hunk 2 above)
Comment (koher) [reply Package.swift]: OK. I'll add `SwiftAST` to `SwiftScript`'s dependencies.

**Comment by koher** [comment]
Fixed. I also removed `Node` because it was used nowhere.

**Review by rintaro — APPROVED** [review]
This looks great!
Thank you 👍

**Pull request merged**